python/adelphi/tests/util/schema_util.py

```diff
@@ -46,22 +50,22 @@ def get_table(keyspace, name):
 	columns_dict["collections"] = column(table, collections_udt.name, "frozen<%s>" % collections_udt.name)
 
 	# set some pk's and ck's
-	columns = list(columns_dict.values())
-	partition_keys = [columns[0], columns[1]]
-	clustering_keys = [columns[2], columns[3]]
+	def data_type_column(idx):
+		return columns_dict[data_type_column_name(idx)]
 
 	table.columns = columns_dict
-	table.partition_key = partition_keys
-	table.clustering_key = clustering_keys
+	table.partition_key = [data_type_column(0), data_type_column(1)]
+	table.clustering_key = [data_type_column(2), data_type_column(3)]
 
 	# indexes
 	indexes = {}
 	# add a regular index
 	regular_index = index(keyspace, name, "regular_index_" + name, None, {"target": "my_column_0"})
 	indexes[regular_index.name] = regular_index
 	# add a CUSTOM index (this must be removed by the anonymizer)
-	custom_index = index(keyspace, name, "custom_index_" + name, "CUSTOM", {"target": "my_column_5", "class_name": "org.apache.cassandra.index.sasi.SASIIndex"})
-	indexes[custom_index.name] = custom_index
+	if sasi:
+		custom_index = index(keyspace, name, "custom_index_" + name, "CUSTOM", {"target": "my_column_5", "class_name": "org.apache.cassandra.index.sasi.SASIIndex"})
+		indexes[custom_index.name] = custom_index
```

Comment:
As mentioned above we must now support schema generation _without_ statements to create SASI indexes since we know they'll fail for at least some Docker images due to a lack of relevant C* config props.